### DISCUSSION TITLE
Using @SendTo annotation with Spring Cloud AWS 3.1.0 SQS Listener

### DISCUSSION THREAD

**Discussion by elsevierstevensn** [awspring/spring-cloud-aws#1047] [Q&A] [answered]
Hello I'm currently updating dependencies on an application which includes a move to version 3.1.0 of Spring Cloud AWS.  Older versions of Spring Cloud AWS supported spring-messaging's `@SendTo` annotation allowing code to be written like:

```
@SqsListener("my-input-queue")
@SendTo("my-output-queue")
public String receiveMessage(String inputMessage) {
   return myService.convert(inputMessage);
}
```

We have noticed after updating that the `@SendTo` annotation no longer has any effect (i.e. the outputMessage is no longer written to the output queue). I think this may be happening because QueueMessagingTemplate from the older version implements DestinationResolvingMessageSendingOperations, while it's replacement SqsTemplate in the newer version does not.

As a workaround I can inject an SqsTemplate into the class and rewrite the method as:

```
@SqsListener("my-input-queue")
public void receiveMessage(String inputMessage) {
    String outputMessage = myService.convert(inputMessage);
    sqsTemplate.send("my-output-queue", outputMessage);
}
```

**My question is "Is there a better workaround or solution?" (such as an equivalent annotation which is valid with the newer version).**  I can't find any mention in the documentation of the fact that `@SendTo` is no longer supported, though I realise the SQS functionality has been completely rewritten to use AWS SDK 2.

**Comment by tomazfernandes** [comment, marked as answer]
Hi @elsevierstevensn.

You're right in that in the rewrite of the solution this feature hasn't been prioritized. Seems like you have a pretty effective workaround, but feel free to open an issue suggesting `@SendTo` implementation.

You can also open a PR for that if you'd like.

Thanks.
